subprojects/core/src/main/java/org/gradle/internal/classpath/InstrumentingClasspathFileTransformer.java

```diff
@@ -146,15 +146,20 @@ private void instrument(File source, File dest) {
 
     private void visitEntries(File source, ClasspathBuilder.EntryBuilder builder) throws IOException, FileException {
         classpathWalker.visit(source, entry -> {
-            if (entry.getName().endsWith(".class")) {
-                ClassReader reader = new ClassReader(entry.getContent());
-                ClassWriter classWriter = new ClassWriter(ClassWriter.COMPUTE_MAXS);
-                Pair<RelativePath, ClassVisitor> chain = transform.apply(entry, classWriter);
-                reader.accept(chain.right, 0);
-                byte[] bytes = classWriter.toByteArray();
-                builder.put(chain.left.getPathString(), bytes);
-            } else {
-                builder.put(entry.getName(), entry.getContent());
+            try {
+                if (entry.getName().endsWith(".class")) {
+                    ClassReader reader = new ClassReader(entry.getContent());
+                    ClassWriter classWriter = new ClassWriter(ClassWriter.COMPUTE_MAXS);
+                    Pair<RelativePath, ClassVisitor> chain = transform.apply(entry, classWriter);
+                    reader.accept(chain.right, 0);
+                    byte[] bytes = classWriter.toByteArray();
+                    builder.put(chain.left.getPathString(), bytes);
+                } else {
+                    builder.put(entry.getName(), entry.getContent());
+                }
+            } catch (Throwable e) {
+                LOGGER.error("Failed to process entry {}", entry.getPath());
+                throw e;
```

Comment:
Please remove the logging and wrap the exception instead. It would make --stacktrace easier to understand.
For instance: `throw new IllegalArgumentException("Can not parse entry " + entry.getPath() + " from " + source, e)`

Reply:
Thanks! I was a bit reluctant initially because each new exception in the chain typically adds one or two screens of output to the already long stacktrace. However, I agree that the connection between the log and the exception may not be obvious.